src/classes/CallableApiParameters.cls

```diff
@@ -148,10 +153,28 @@ public class CallableApiParameters {
         return null;
     }
 
+    /**
+     * @description Retrieve the specified Parameter as a Date value
+     * @param paramName Map key for the params collection
+     * @return Date
+     */
+    public Date getDate(String paramName) {
+        validateRequiredParameter('Date', paramName);
+        try {
+            if (params.get(paramName) != null) {
```

Comment:
No need to check the param has the value since it is already checked in the `validateRequiredParameter`.